app/src/main/java/protect/card_locker/Utils.java

```diff
@@ -672,11 +673,12 @@ public static void postPatchColors(AppCompatActivity activity) {
         activity.getTheme().resolveAttribute(android.R.attr.colorBackground, typedValue, true);
         activity.findViewById(android.R.id.content).setBackgroundColor(typedValue.data);
 
-        if (Build.VERSION.SDK_INT >= 27) {
-            View decorView = activity.getWindow().getDecorView();
-            WindowInsetsControllerCompat wic = new WindowInsetsControllerCompat(activity.getWindow(), decorView);
+        Window window = activity.getWindow();
+        if (window != null && Build.VERSION.SDK_INT >= 27) {
+            View decorView = window.getDecorView();
+            WindowInsetsControllerCompat wic = new WindowInsetsControllerCompat(window, decorView);
             wic.setAppearanceLightNavigationBars(!isDarkModeEnabled(activity));
-            activity.getWindow().setNavigationBarColor(typedValue.data);
+            window.setNavigationBarColor(typedValue.data);
         }
```

Comment:
I'd switch this order because the variable is unused on older Android versions. Not a huge deal, but easier to simplify later

```
        if (Build.VERSION.SDK_INT >= 27) {
            Window window = activity.getWindow();
            if (window != null) {
                View decorView = window.getDecorView();
                WindowInsetsControllerCompat wic = new WindowInsetsControllerCompat(window, decorView);
                wic.setAppearanceLightNavigationBars(!isDarkModeEnabled(activity));
                window.setNavigationBarColor(typedValue.data);
            }
        }
```

Reply:
I figured the call to `getWindow()` was likely inexpensive and preferred to avoid another level of nesting. But yeah, I did consider that. Will change.